components/count-badge/count-badge.js

```diff
@@ -48,14 +50,45 @@ class CountBadge extends RtlMixin(LitElement) {
 			 */
 			hideZero: {
 				type: Boolean,
-				reflect: true,
 				attribute: 'hide-zero'
 			},
+			/**
+			 * The text that will show as an aria-label on the badge. NOTE: Only the text will be read by screen-readers (not the number).
+			 * @type {string}
+			 */
+			text: {
+				type: String,
+				attribute: 'text'
+			},
```

Comment:
You don't need to set `attribute` if the desired name matches the property name.

```suggestion
			text: {
				type: String
			},
```